### PR TITLE
fix to build system to limit number of threads #115

### PR DESCRIPTION
**Pull request by nickjbrowning**
Now uses at most 8 threads.

### CODE REVIEW

**Comment by frostedoyster** [comment]
@Luthaf what do you think about this PR?
Part of me feels that Marcel should have just requested more memory on his HPC node

**Comment by Luthaf** [comment]
I think this is worth it: anything that include torch takes a lot of memory to compile, and sometime the front node has limits on how much memory a single user can use. Also, we don't have that many files to compile, so using more than 8 threads won't really help much.

I'm not sure how much RAM compilation with 8 threads would take though, do we know that this is low enough to fix the issue?

**Comment by nickjbrowning** [comment]
I've added two environment variables that modify the pip build process:

see: https://github.com/lab-cosmo/sphericart/issues/115#issuecomment-2225053275

**Comment by frostedoyster** [comment]
Are we sure it's a good idea to have an undocumented environment variable? I think `min(cpu_count, 8)` is sufficient and simple enough

**Comment by Luthaf** [comment]
>  I think min(cpu_count, 8) is sufficient 

I'm not sure. We barely have 8 files to compile, so if the build fails with `make -j` it will also fail with `make -j8`. We could do a lower default limit (2-3) if you prefer, or query available RAM and use `min(cpu_count, ram/4GiB)` (4GiB is close to what I saw a single compiler process take when building code that `#include <torch/...>`.

**Comment by frostedoyster** [comment]
Querying the RAM is probably the proper fix for the original issue, so I will try to implement it

**Comment by frostedoyster** [comment]
EDIT: it seems that there isn't a reliable way to get the RAM from the Python standard library, we're now doing two jobs

File: sphericart-jax/setup.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import os
 import subprocess
 import sys
+import multiprocessing
```
Comment (Luthaf) [inline sphericart-jax/setup.py:4]:
```suggestion
```

Comment (frostedoyster) [reply sphericart-jax/setup.py:4]: Hmmm thanks, it seems that the linter didn't catch that, I will investigate

**Pull request merged**